indico/modules/events/abstracts/util.py

```diff
@@ -345,24 +326,31 @@ def get_events_with_abstract_reviewer_convener(user, dt=None):
     """
     data = defaultdict(set)
     # global reviewer/convener
-    mapping = {'global_abstract_reviewer_for_events': 'abstract_reviewer',
-               'global_convener_for_events': 'track_convener'}
-    for rel, role in mapping.iteritems():
-        query = (Event.query.with_parent(user, rel)
-                 .filter(Event.ends_after(dt), ~Event.is_deleted)
-                 .options(load_only('id')))
-        for event in query:
-            data[event.id].add(role)
-    # track reviewer/convener
-    mapping = {'abstract_reviewer_for_tracks': 'abstract_reviewer',
-               'convener_for_tracks': 'track_convener'}
-    for rel, role in mapping.iteritems():
-        query = (Track.query.with_parent(user, rel)
-                 .join(Track.event)
-                 .filter(Event.ends_after(dt), ~Event.is_deleted)
-                 .options(load_only('event_id')))
-        for track in query:
-            data[track.event_id].add(role)
+
+    event_query = (user.in_event_acls
+                   .join(Event)
+                   .options(noload('user'), noload('local_group'), load_only('event_id', 'permissions'))
+                   .filter(Event.ends_after(dt), ~Event.is_deleted))
+    for principal in event_query:
+        roles = data[principal.event_id]
+        if 'review_all_abstracts' in principal.permissions:
+            roles.add('abstract_reviewer')
+        if 'convene_all_abstracts' in principal.permissions:
+            roles.add('track_convener')
+
+    query = (user.in_track_acls
+             .options(load_only('track_id', 'permissions'))
+             .options(noload('*'))
+             .options(contains_eager(TrackPrincipal.track).load_only('event_id'))
```

Comment:
You aren't using the track for any filtering, so no need for the explicit join + contains_eager - you can just `joinedload()` it instead